metadata/metadata_test.go

```diff
@@ -385,5 +404,197 @@ func TestTargetFilesCustomField(t *testing.T) {
 	targets.Signed.Targets["testTarget"] = targetFile
 	targetsJSON, err := targets.ToBytes(false)
 	assert.NoError(t, err)
-	assert.Equal(t, []byte("{\"signed\":{\"_type\":\"targets\",\"expires\":\"2030-08-15T14:30:45.0000001Z\",\"spec_version\":\"1.0.31\",\"targets\":{\"testTarget\":{\"custom\":{\"test\":true},\"hashes\":{},\"length\":0}},\"version\":1},\"signatures\":[]}"), targetsJSON)
+	assert.Equal(t, exampleTargetsBytes, targetsJSON)
+}
+
+func TestFromBytes(t *testing.T) {
+	safeExpiry := getSafeExpiry()
+	root := Root(safeExpiry)
+	assert.Equal(t, safeExpiry, root.Signed.Expires)
+
+	_, err := root.FromBytes(exampleRootBytes)
+	if assert.NoError(t, err) {
+		exampleRootBytesExpire := "2030-08-15T14:30:45.0000001Z"
+		expectedNewTime, err := time.Parse(time.RFC3339, exampleRootBytesExpire)
+		if assert.NoError(t, err) {
+			assert.Equal(t, expectedNewTime, root.Signed.Expires)
+		}
+	}
+}
+
+func TestToByte(t *testing.T) {
+	rootBytesExpireStr := "2030-08-15T14:30:45.0000001Z"
+	rootBytesExpire, err := time.Parse(time.RFC3339, rootBytesExpireStr)
+	if assert.NoError(t, err) {
+		root := Root(rootBytesExpire)
+		unrecognizedField := map[string]any{"test": "true"}
+		root.Signed.UnrecognizedFields = unrecognizedField
+		rootBytes, err := root.ToBytes(false)
+		assert.NoError(t, err)
+		assert.Equal(t, exampleRootBytes, rootBytes)
+	}
```

Comment:
I believe this is already covered in one of the existing [tests](https://github.com/rdimitrov/go-tuf-metadata/blob/f42c9df46a15684ea4bbcaf32c5e0f7d1b17d3ba/metadata/metadata_test.go#L337) so perhaps we should either extend this test case as in its current state there's no practical value in it.